elytron/src/test/java/org/wildfly/extension/elytron/RealmsTestCase.java

```diff
@@ -145,6 +145,18 @@ public void testOAuth2Realm() throws Exception {
         Assert.assertNotNull(securityRealm);
     }
 
+    @Test
+    public void testAggregateRealm() throws Exception {
+        KernelServices services = super.createKernelServicesBuilder(new TestEnvironment()).setSubsystemXmlResource("realms-test.xml").build();
```

Comment:
I think it would be good to expand this test to check if the configured principal transformer actually gets used properly, similar to the tests you have in the Elytron testsuite.

Reply:
Fixed

Reply:
Thanks @ashley-abdelsayed98! I think it would be good to rename this test to ```testAggregateRealmWithPrincipalTransformer``` and then also add a test for the case where there isn't a principal transformer configured.

Reply:
added